app/models/message.rb

```diff
@@ -189,11 +189,24 @@ def send_reply
     attachments.blank? ? ::SendReplyJob.perform_later(id) : ::SendReplyJob.set(wait: 2.seconds).perform_later(id)
   end
 
-  def reopen_conversation
+  def update_conversation_status
     return if conversation.muted?
     return unless incoming?
 
-    conversation.open! if conversation.resolved? || conversation.snoozed?
+    conversation.open! if conversation.snoozed?
+
+    if conversation.resolved?
+      if conversation.inbox.active_bot?
+        conversation.pending!
+      else
+        conversation.open!
+      end
+    end
+  end
+
+  def mark_bot_conversation_pending
```

Comment:
@smartdev58 are we using this method anywhere ?

Reply:
@sojan-official  The method is being used here: https://github.com/chatwoot/chatwoot/blob/e1190fd9bf171a3e95215bacb9bbeb217d7b4a1a/app/models/conversation.rb#L92

Reply:
I think this must be a mistake while I merged the code. I have updated the method above. We could remove this.

Reply:
@smartdev58 I can update the method as per the suggestion and get it merged.

Reply:
@pranavrajs Thank you. I am done with other comments here.